### PR TITLE
Update HTSLIB dependency and fix missing include

### PR DESCRIPTION
**Pull request by jgarciamesa**
Add `include fstream` on `io/file.h` for usage of fstream class.

### CODE REVIEW

**Comment by reedacartwright** [comment]
Jenkins, retest this please

**Comment by reedacartwright** [comment]
Jenkins, retest this please.

**Comment by jgarciamesa** [comment]
I will include the changes to HTSlib on this pull request. I checked and version 1.3, which the version that is currently build as an external project if needed, works fine for all the current test cases. However, the following version 1.3.1 includes numerous VCF parsing fixes. I was wondering if we should request 1.3.1 to be the minimum version to run DNG and then be the one we build if needed.

**Comment by reedacartwright** [comment]
Update supported version of htslib to 1.3.1

--
Reed A. Cartwright, PhD
Barrett Honors Faculty
Assistant Professor of Genomics, Evolution, and Bioinformatics
School of Life Sciences
Human and Comparative Genomics Laboratory
The Biodesign Institute
Arizona State University
==================
Availability: http://links.asu.edu/CartwrightCalendar
Address: The Biodesign Institute, PO Box 875301, Tempe, AZ 85287-5301 USA
Packages: The Biodesign Institute, 1001 S. McAllister Ave, Tempe, AZ
85287-5301 USA
Office: Biodesign B-220C, 1-480-965-9949
Website: http://cartwrig.ht/

**Comment by reedacartwright** [comment]
Jenkins, retest this please

File: Modules/FindHTSLIB.cmake [hunk 1]
```diff
@@ -9,6 +9,14 @@ libfind_pkg_detect(HTSLIB htslib FIND_PATH htslib/hts.h FIND_LIBRARY hts)
 if(NOT HTSLIB_VERSION)
   if(HTSLIB_PKGCONF_VERSION)
     set(HTSLIB_VERSION "${HTSLIB_PKGCONF_VERSION}")
+  else()
+    set(cmd "-c")
+    set(args "htsfile --version | grep -o -E '[0-9.]+' | head -1")
```
Comment (reedacartwright) [inline Modules/FindHTSLIB.cmake]: Use FIND_PROGRAM to locate the htsfile.
Don't use GREP the parse the output, parse in in the CMAKE file.

File: Modules/FindHTSLIB.cmake [hunk 2]
```diff
@@ -9,6 +9,14 @@ libfind_pkg_detect(HTSLIB htslib FIND_PATH htslib/hts.h FIND_LIBRARY hts)
...
+    execute_process(
```
Comment (reedacartwright) [inline Modules/FindHTSLIB.cmake]: Execute htsfile --version directly,not via bash.

File: Modules/FindHTSLIB.cmake [hunk 3]
```diff
@@ -9,6 +9,14 @@ libfind_pkg_detect(HTSLIB htslib FIND_PATH htslib/hts.h FIND_LIBRARY hts)
...
+      COMMAND bash ${cmd} ${args}
+      OUTPUT_VARIABLE version
+    )
+    set(HTSLIB_VERSION ${version})
```
Comment (reedacartwright) [inline Modules/FindHTSLIB.cmake]: output to variable `hts_version` instead of simply `version`

File: Tests/Unit/testing.h [hunk 4]
```diff
@@ -151,21 +151,21 @@ struct AutoTempFile {
 // function to create a 
 
 namespace {
-template<typename It>
+template<typename T>
```
Comment (reedacartwright) [inline Tests/Unit/testing.h]: Use `It` instead of `T` in the template

File: Tests/Unit/testing.h [hunk 5]
```diff
@@ -151,21 +151,21 @@ struct AutoTempFile {
...
+    using iterator = T;
+    using const_iterator = T;
+    using value_type = typename std::iterator_traits<T>::value_type;
+    using reference = typename std::iterator_traits<T>::reference;
+    using difference_type = typename std::iterator_traits<T>::difference_type;
+    using pointer = typename std::iterator_traits<T>::pointer;
+    using iterator_category = typename std::iterator_traits<T>::iterator_category;
     
-    test_range(It const& b, It const &e) : begin_{b}, end_{e} { }
+    test_range(T b, T e) : begin_{b}, end_{e} { }
```
Comment (reedacartwright) [inline Tests/Unit/testing.h]: Use the `It const& ` version

**Comment by reedacartwright** [comment]
Jenkins, retest this please

**Pull request merged**